### PR TITLE
Drop-ins Initialization

### PR DESCRIPTION
**Pull request by fnhipster**
# Description

Initializes DropIns

## Pending 

- [x] Install vendors
- [x] Initialization file

## Test URLs

- https://dropins-initialization--aem-boilerplate-commerce--hlxsites.hlx.live/
- https://dropins-initialization--aem-boilerplate-commerce--hlxsites.hlx.live/cart
- https://dropins-initialization--aem-boilerplate-commerce--hlxsites.hlx.live/checkout
- https://dropins-initialization--aem-boilerplate-commerce--hlxsites.hlx.live/order-confirmation?orderRef=abc
- https://dropins-initialization--aem-boilerplate-commerce--hlxsites.hlx.live/products/crown-summit-backpack/24-MB03

### CODE REVIEW

**Review by sirugh — APPROVED** [review]

File: scripts/dropins.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import * as product from '@dropins/storefront-pdp/api.js';
+import * as checkout from '@dropins/storefront-checkout/api.js';
+import * as orderConfirmation from '@dropins/storefront-order-confirmation/api.js';
+
+// Libs
+import { getConfigValue } from './configs.js';
+
+export default async function initializeDropins() {
+  // Set Commerce Mesh endpoint
+  mesh.setEndpoint(await getConfigValue('commerce-core-endpoint'));
```
Comment (sirugh) [inline scripts/dropins.js:17]: Calling the import `mesh` can be confusing - as it's unrelated to actual API Mesh.

File: scripts/dropins.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  // Initializers
+  initializers.register(product.initialize, {});
+  initializers.register(cart.initialize, {});
+  initializers.register(checkout.initialize, {});
+  initializers.register(orderConfirmation.initialize, {});
```
[the remark this answers is not in the document]
Comment (fnhipster) [reply scripts/dropins.js:23]: @sirugh I've been bouncing back and forth with that. You are right, Maybe This is where you can initialize a global drop-in like Auth and Cart but the rest is initialized in their block.

File: blocks/commerce-checkout/commerce-checkout.js [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
-    // TODO:
-    // cartId: store.getCartId(),
+export default async function decorate(block) {
+  // Listen for order confirmation and redirect to order confirmation page
+  events.on('checkout/order', (data) => {
+    window.location.replace(`/order-confirmation?orderRef=${data.masked_order_id}`);
   });
 
-  checkoutRenderer.render(Checkout, {
+  return provider.render(Checkout, {
```
Comment (sirugh) [inline blocks/commerce-checkout/commerce-checkout.js:26]: Is this meant to be standard practice for all drop-in rendering?

File: scripts/dropins.js (hunk 1 above)
Comment (fnhipster) [reply scripts/dropins.js]: Fixed. Added a space for global initialization like Cart and Auth and blocks initialize their dropins

File: scripts/dropins.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (fnhipster) [reply scripts/dropins.js]: Fixed

File: blocks/commerce-checkout/commerce-checkout.js (hunk 3 above)
Comment (fnhipster) [reply blocks/commerce-checkout/commerce-checkout.js:26]: yes

**Pull request merged**